pkg/devtronResource/history/deployment/cdPipeline/DeploymentHistoryService.go

```diff
@@ -17,19 +18,22 @@ type DeploymentHistoryService interface {
 type DeploymentHistoryServiceImpl struct {
 	logger                              *zap.SugaredLogger
 	cdHandler                           pipeline.CdHandler
+	imageTaggingReadService             read.ImageTaggingReadService
 	imageTaggingService                 pipeline.ImageTaggingService
```

Comment:
remove old import

Reply:
There is this method, using `imageTaggingService`. Ideally `GetProdEnvByCdPipelineId` should be the part of `CdPipelineReadService`
```go
prodEnvExists, err := impl.imageTaggingService.GetProdEnvByCdPipelineId(req.PipelineId)
```